volumes/miovision/sql/function-find_gaps.sql

```diff
@@ -14,10 +14,36 @@ DECLARE tot_gaps integer;
 BEGIN
 
 -- FIRST, FIND ALL GAPS
-WITH ful AS (
-SELECT generate_series(start_date, end_date, interval '1 minute')::timestamp without time zone 
-	AS datetime_bin)
-, grp AS (
+WITH wkdy_lookup(period, isodow) AS (
+         VALUES ('Weekday'::text,'[1,6)'::int4range), ('Weekend'::text,'[6,8)'::int4range)
+), mio AS (
+ SELECT volumes.intersection_uid,
+    datetime_bin(volumes.datetime_bin, 60) AS hourly_bin,
+    sum(volumes.volume) AS vol
+   FROM miovision_api.volumes
+  WHERE volumes.datetime_bin > (GREATEST(end_date::timestamp without time zone, '2019-03-01'::timestamp without time zone) - '60 days'::interval) AND
+        volumes.datetime_bin <= GREATEST(end_date::timestamp without time zone, '2019-03-01'::timestamp without time zone)
+  GROUP BY volumes.intersection_uid, (datetime_bin(volumes.datetime_bin, 60))
+), gapsize_lookup AS (
+ SELECT mio.intersection_uid,
+    d.period,
+    mio.hourly_bin::time without time zone AS time_bin,
+    avg(mio.vol) AS avg_vol,
+        CASE
+            WHEN avg(mio.vol) < 100::numeric THEN 20
+            WHEN avg(mio.vol) >= 100::numeric AND avg(mio.vol) < 500::numeric THEN 15
+            WHEN avg(mio.vol) >= 500::numeric AND avg(mio.vol) < 1500::numeric THEN 10
+            WHEN avg(mio.vol) > 1500::numeric THEN 5
+            ELSE NULL::integer
+        END AS gap_size
```

Comment:
this should be called `gap_tolerance` so it's less ambiguous with `gap_size` further down.

Reply:
Done